oval_graph/oval_node.py

```diff
@@ -16,60 +16,69 @@ class OvalNode():
-        or, one etc... and for value true, false, error etc...
+        value (str): value of node for operator and, or, one etc... and for value true,
+        false, error etc...
+        negation (bool): value indicating whether the node is negated
+        comment (str): some comment about node
+        tag (str): tag specifies if the node represents OVAL test,
+        OVAL definition or XCCDF rule
+        test_result_details (dict|None): information about test
         children ([OvalNode]): children of node
     '''
 
-    def __init__(
-            self,
-            node_id,
-            input_node_type,
-            input_value,
-            input_negation,
-            comment,
-            tag,
-            test_result_details,
-            children=None
-    ):
-        self.comment = comment
-        self.node_id = node_id
-        self.negation = self.validate_negation(input_negation)
-        self.node_type = self.validate_type(input_node_type)
-        self.value = self.validate_type_and_value(input_value)
+    def __init__(self, **kwargs):
+        try:
+            assert 'node_id' in kwargs
+            assert 'node_type' in kwargs
+            assert 'value' in kwargs
+            self.node_id = kwargs.get('node_id')
+            self.node_type = self._validate_type(kwargs.get('node_type'))
+            self.value = self._validate_value(kwargs.get('value'))
+
+            self.negation = self._validate_negation(
+                kwargs.get('negation', False))
+        except AssertionError:
+            raise ValueError(
+                "err- miss require argument or bad value of require argument!")
+
+        self.comment = kwargs.get('comment', None)
+        self.tag = kwargs.get('tag', None)
+        self.test_result_details = kwargs.get('test_result_details', None)
+
+        self._check_missing_children_for_operator(
+            kwargs.get('children', None))
+        input_children = kwargs.get('children', None)
         self.children = []
-        self.validate_children(children)
-        self.tag = tag
-        self.test_result_details = test_result_details
-        if children:
-            for child in children:
-                self.add_child(child)
-
-    def validate_negation(self, input_negation):
-        if not isinstance(input_negation, bool):
-            raise ValueError("err- negation is bool (only True or False)")
+        if input_children:
+            for child in input_children:
+                self._add_child(child)
+
+    def _validate_negation(self, input_negation):
+        assert isinstance(input_negation, bool)
         return input_negation
 
-    def validate_type(self, input_node_type):
+    def _validate_type(self, input_node_type):
         node_type = input_node_type.lower()
-        if node_type != "value" and node_type != "operator":
-            raise ValueError("err- unknown type")
+        assert node_type == "value" or node_type == "operator"
```

Comment:
You can use `node_type in ("value", ...)`, and in this case, it is better to raise a [TypeError](https://docs.python.org/3/library/exceptions.html#TypeError).